kv/kv.go

```diff
@@ -0,0 +1,155 @@
+package kv
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/jordan-rash/wasmcloud-go/models"
+	"github.com/nats-io/nats.go"
+	core "github.com/wasmcloud/interfaces/core/tinygo"
+)
+
+const (
+	LINKDEF_PREFIX = "LINKDEF_"
+	CLAIMS_PREFIX  = "CLAIMS_"
+	BUCKET_PREFIX  = "LATTICEDATA_"
+)
+
+var (
+	DEFAULT_LEAFNOTE_OPTIONS nats.JSOpt = nats.PublishAsyncMaxPending(256)
+)
+
+func GetKVStore(nc *nats.Conn, latticePrefix, jsDomain string) (nats.KeyValue, error) {
+	var js nats.JetStreamContext
+	var err error
+
+	if jsDomain != "" {
+		js, err = nc.JetStream(nats.Domain(jsDomain), DEFAULT_LEAFNOTE_OPTIONS)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		js, err = nc.JetStream(DEFAULT_LEAFNOTE_OPTIONS)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	bucket := fmt.Sprintf("%s%s", BUCKET_PREFIX, latticePrefix)
+
+	return js.KeyValue(bucket)
+}
+
+func GetClaims(store nats.KeyValue) (*models.GetClaimsResponse, error) {
+	claims := models.CtlKVList{}
+	entries, err := store.Keys()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, c := range entries {
+		if strings.HasPrefix(c, CLAIMS_PREFIX) {
+			entry, err := store.Get(c)
+			if err != nil {
+				return nil, err
+			}
+
+			d := models.KeyValueMap{}
+			err = json.Unmarshal(entry.Value(), &d)
+			if err != nil {
+				return nil, err
+			}
+
+			claims = append(claims, d)
+			if err != nil {
```

Comment:
Nit: no new `err` here, so this check looks redundant?

Reply:
How embarrassing 🙈